### PR TITLE
Set a removal policy to ENI

### PR DESCRIPTION
**Pull request by magicien**
*Issue #, if available:*

None.

Issue: The ENI was replaced with a new one when I changed VPC settings, and my Unity license doesn't work after that.

*Description of changes:*

This pull request sets a removal polity to ENI to avoid the accidental deletion.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cdk/lib/unity-license-server-stack.ts [hunk 1]
```diff
@@ -56,6 +56,10 @@ export class UnityLicenseServerStack extends cdk.Stack {
         },
       ],
     });
+    eni.applyRemovalPolicy(RemovalPolicy.RETAIN, {
+      applyToUpdateReplacePolicy: true,
+      default: cdk.RemovalPolicy.RETAIN,
+    });
```
Comment (tmokmss) [inline cdk/lib/unity-license-server-stack.ts]: It should not need the 2nd argument in this case.

```suggestion
    eni.applyRemovalPolicy(RemovalPolicy.RETAIN);
```

**Comment by tmokmss** [comment]
Hi @magicien, thank you the PR! I'll merge this when the comment is addressed. I'll also add a warning about replacing the ENI, and how it can be avoided.

As for your Unity license, I believe Unity support can help you.

**Comment by tmokmss** [comment]
Thanks!

**Comment by magicien** [comment]
@tmokmss Thanks! I merged the suggested change.
I sent a message to Unity, and they responded so quickly that the license recovered in the day.

**Comment by tmokmss** [comment]
Glad to hear that 😄  Give us more feedback if you find anything, thanks!

**Pull request merged**